### PR TITLE
Additions to agent docs

### PR DESCRIPTION
**Pull request by enyst**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Added docstrings for `agent_controller` and `state`, adjusted and added to the agent Readme.

### CODE REVIEW

File: agenthub/README.md [hunk 1]
```diff
@@ -70,3 +91,50 @@ def step(self, state: "State") -> "Action"
 
 `step` moves the agent forward one step towards its goal. This probably means
 sending a prompt to the LLM, then parsing the response into an `Action`.
+
+## Agent Delegation
+
+        OpenDevin is a multi-agentic system. Agents can delegate tasks to other agents, whether
```
Comment (xingyaoww) [inline agenthub/README.md:97]: Should we remove the extra indentation here?

Comment (enyst) [reply agenthub/README.md]: Fixed

File: opendevin/controller/agent_controller.py [hunk 2]
```diff
@@ -111,6 +111,9 @@ def __init__(
             self.agent_task = asyncio.create_task(self._start_step_loop())
 
     async def close(self):
+        """
```
Comment (tobitege) [inline opendevin/controller/agent_controller.py:114]: nit: no newline at start

File: agenthub/README.md [hunk 3]
```diff
@@ -70,3 +91,52 @@ def step(self, state: "State") -> "Action"
...
+the `CodeActAgent` might delegate to the `BrowsingAgent` to answer questions that involve browsing
+the web. The Delegator Agent forwards tasks to micro-agents, such as 'RepoStudyAgent' to study a repo,
+or 'VerifierAgent' to verify a task completion.
+
+### Understanding the terminology
+
+A `task` is an end-to-end conversation between OpenDevin (the whole system) and the user,
+which might involve one or more inputs from the user. It starts with an initial input
+(typically a task statement) from the user, and ends with either an `AgentFinishAction`
+initiated by the agent, or an error.
```
Comment (tobitege) [inline agenthub/README.md:108]: or just end in STOPPED mode

File: opendevin/controller/agent_controller.py [hunk 4]
```diff
@@ -137,6 +140,9 @@ async def report_error(self, message: str, exception: Exception | None = None):
         self.event_stream.add_event(ErrorObservation(message), EventSource.AGENT)
 
     async def _start_step_loop(self):
+        """
```
Comment (tobitege) [inline opendevin/controller/agent_controller.py:143]: nit: newline, see below occurences as well

File: opendevin/controller/agent_controller.py [hunk 5]
```diff
@@ -251,10 +272,29 @@ async def set_agent_state_to(self, new_state: AgentState):
...
+
+        Returns:
+            AgentState: The current state of the agent.
+        """
         return self.state.agent_state
 
     async def start_delegate(self, action: AgentDelegateAction):
+        """Start a delegate agent to handle a subtask.
+
+        OpenDevin is a multi-agentic system. A `task` is a conversation between
```
Comment (tobitege) [inline opendevin/controller/agent_controller.py:286]: maybe shorten this like "See also step() method."?

**Review by tobitege — APPROVED** [review]
Only tiny nits. Helpful additions! LGTM

File: opendevin/controller/agent_controller.py (hunk 5 above)
Comment (enyst) [reply opendevin/controller/agent_controller.py:286]: I don't know, the text here was in State, AND much more. It's already shortened. 😅 It just seems like a good idea to have some quick view of the delegate process in the code itself? They're not too trivial. It's true that most is in the readme now though.

Comment (enyst) [reply opendevin/controller/agent_controller.py:279]: I think it looks good, looking at the whole file in the editor. Let's go with it and we'll see if we hate it.

**Pull request merged**